code/modules/hydroponics/hydroponics.dm

```diff
@@ -75,6 +75,7 @@
 	plant_hud_set_water()
 
 /obj/machinery/hydroponics/Destroy()
+	remove_from_all_data_huds()
```

Comment:
this is another option proc wise but i think its a permob thing https://github.com/ParadiseSS13/Paradise/blob/18f4aa18b98fcf8503f7173d3e294d581aa2acff/code/datums/hud.dm#L38